In [ ]:
import pandas as pd
import numpy as np

In [ ]:
data = pd.read_csv('/content/drive/My Drive/전산통계/brain_size.csv', sep=';', na_values='.')

In [ ]:
data.head()

Unnamed: 0  Gender  FSIQ  VIQ  PIQ  Weight  Height  MRI_Count
0           1  Female   133  132  124   118.0    64.5     816932
1           2    Male   140  150  124     NaN    72.5    1001121
2           3    Male   139  123  150   143.0    73.3    1038437
3           4    Male   133  129  128   172.0    68.8     965353
4           5  Female   137  132  134   147.0    65.0     951545

In [ ]:
data.fillna(data.mean())

Unnamed: 0  Gender  FSIQ  VIQ  PIQ      Weight     Height  MRI_Count
0            1  Female   133  132  124  118.000000  64.500000     816932
1            2    Male   140  150  124  151.052632  72.500000    1001121
2            3    Male   139  123  150  143.000000  73.300000    1038437
3            4    Male   133  129  128  172.000000  68.800000     965353
4            5  Female   137  132  134  147.000000  65.000000     951545
5            6  Female    99   90  110  146.000000  69.000000     928799
6            7  Female   138  136  131  138.000000  64.500000     991305
7            8  Female    92   90   98  175.000000  66.000000     854258
8            9    Male    89   93   84  134.000000  66.300000     904858
9           10    Male   133  114  147  172.000000  68.800000     955466
10          11  Female   132  129  124  118.000000  64.500000     833868
11          12    Male   141  150  128  151.000000  70.000000    1079549
12          13    Male   135  129  124  155.000000  69.000000     924059
13          14  Female   140  120  147  155.000000  70.500000     856472
14          15  Female    96  100   90  146.000000  66.000000     878897
15          16  Female    83   71   96  135.000000  68.000000     865363
16          17  Female   132  132  120  127.000000  68.500000     852244
17          18    Male   100   96  102  178.000000  73.500000     945088
18          19  Female   101  112   84  136.000000  66.300000     808020
19          20    Male    80   77   86  180.000000  70.000000     889083
20          21    Male    83   83   86  151.052632  68.525641     892420
21          22    Male    97  107   84  186.000000  76.500000     905940
22          23  Female   135  129  134  122.000000  62.000000     790619
23          24    Male   139  145  128  132.000000  68.000000     955003
24          25  Female    91   86  102  114.000000  63.000000     831772
25          26    Male   141  145  131  171.000000  72.000000     935494
26          27  Female    85   90   84  140.000000  68.000000     798612
27          28    Male   103   96  110  187.000000  77.000000    1062462
28          29  Female    77   83   72  106.000000  63.000000     793549
29          30  Female   130  126  124  159.000000  66.500000     866662
30          31  Female   133  126  132  127.000000  62.500000     857782
31          32    Male   144  145  137  191.000000  67.000000     949589
32          33    Male   103   96  110  192.000000  75.500000     997925
33          34    Male    90   96   86  181.000000  69.000000     879987
34          35  Female    83   90   81  143.000000  66.500000     834344
35          36  Female   133  129  128  153.000000  66.500000     948066
36          37    Male   140  150  124  144.000000  70.500000     949395
37          38  Female    88   86   94  139.000000  64.500000     893983
38          39    Male    81   90   74  148.000000  74.000000     930016
39          40    Male    89   91   89  179.000000  75.500000     935863

In [ ]:
from statsmodels.formula.api import ols

model = ols('VIQ ~ Gender + Weight + Height + MRI_Count', data).fit()
print(model.summary())

                            OLS Regression Results                            
Dep. Variable:                    VIQ   R-squared:                       0.249
Model:                            OLS   Adj. R-squared:                  0.158
Method:                 Least Squares   F-statistic:                     2.733
Date:                Wed, 11 Nov 2020   Prob (F-statistic):             0.0455
Time:                        03:12:29   Log-Likelihood:                -167.03
No. Observations:                  38   AIC:                             344.1
Df Residuals:                      33   BIC:                             352.2
Df Model:                           4                                         
Covariance Type:            nonrobust                                         
                     coef    std err          t      P>|t|      [0.025      0.975]
----------------------------------------------------------------------------------
Intercept        169.7719     90.054      1.

Gender 변수의 coef가 10.15로 남자일 경우 VIQ가 10정도 높다는 것을 의미하는데 Gender 변수에 대한 p-value가 0.358로 높게 나왔다. 즉, 성별에 따라 VIQ가 영향을 받지 않는 것으로 볼 수 있다. 
MRI_Count의 p-value값은 유의미 하지만 coef의 값이 작아 큰 의미가 없어 보인다. 

In [ ]:
model2 = ols('VIQ ~ FSIQ + PIQ', data).fit()
print(model2.summary())

                            OLS Regression Results                            
Dep. Variable:                    VIQ   R-squared:                       0.985
Model:                            OLS   Adj. R-squared:                  0.984
Method:                 Least Squares   F-statistic:                     1179.
Date:                Wed, 11 Nov 2020   Prob (F-statistic):           3.12e-34
Time:                        03:15:30   Log-Likelihood:                -99.322
No. Observations:                  40   AIC:                             204.6
Df Residuals:                      37   BIC:                             209.7
Df Model:                           2                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept     17.6592      2.435      7.253      0.0

FSIQ가 높은 사람이 VIQ가 1.69정도 높고 PIQ가 높은 사람이 VIQ가 0.87정도 낮은 것으로 볼 수 있다. 

VIQ^(hat) = 17.6592 + 1.6915 * FSIQ - 0.8755 * PIQ